core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -451,8 +537,10 @@ private void validateNoNewDeletesForDataFiles(
   }
 
   /**
-   * Validates that no delete files matching a filter have been added to the table since a starting
-   * snapshot.
+   * ToDo: Remove after branch writing implementation complete
```

Comment:
We can remove the existing validate after we implement the different write operations because we will always have an ending snapshot.